src/mango/src/mango_selector_text.erl

```diff
@@ -142,6 +142,10 @@ convert(Path, {[{<<"$exists">>, ShouldExist}]}) ->
         true -> FieldExists;
         false -> {op_not, {FieldExists, false}}
     end;
+convert(Path, {[{<<"$beginsWith">>, Arg}]}) when is_binary(Arg) ->
+    Suffix = <<"*">>,
+    PrefixSearch = value_str(<<Arg/binary, Suffix/binary>>),
+    {op_field, {make_field(Path, Arg), PrefixSearch}};
```

Comment:
It is almost there.  The problem with this version is that it will quote the `*` (wildcard) therefore neutralize that.  Here is an alternative that fixed `03-operator-test.OperatorTextTests.test_beginswith` for me:

```erlang
convert(Path, {[{<<"$beginsWith">>, Arg}]}) when is_binary(Arg) ->
    Prefix = mango_util:lucene_escape_query_value(Arg),
    Suffix = <<"*">>,
    PrefixSearch = <<Prefix/binary, Suffix/binary>>,
    {op_field, {make_field(Path, Arg), PrefixSearch}};
```

Reply:
On that note, I would recommend to include spaces or other special characters (subject to escaping) in the strings used in the tests.  For example (note the space after `"New"`):

```python
         def test_beginswith(self):
             docs = self.db.find({"location.state": {"$beginsWith": "New "}})
             self.assertEqual(len(docs), 2)
             self.assertUserIds([2, 10], docs)
```

Reply:
thanks - I included the fix and added some covering test cases for escaped characters.